platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClaimsHandler.java

```diff
@@ -0,0 +1,259 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.claims.attributequery;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.cxf.rt.security.claims.ClaimCollection;
+import org.apache.cxf.sts.claims.ClaimsHandler;
+import org.apache.cxf.sts.claims.ClaimsParameters;
+import org.apache.cxf.sts.claims.ProcessedClaim;
+import org.apache.cxf.sts.claims.ProcessedClaimCollection;
+import org.opensaml.saml2.core.Assertion;
+import org.opensaml.saml2.core.Attribute;
+import org.opensaml.saml2.core.AttributeStatement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
+
+import ddf.security.PropertiesLoader;
+import ddf.security.common.audit.SecurityLogger;
+import ddf.security.samlp.SimpleSign;
+
+public class AttributeQueryClaimsHandler implements ClaimsHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AttributeQueryClaimsHandler.class);
+
+    protected static final String ERROR_RETRIEVING_ATTRIBUTES = "Error retrieving attributes from external attribute store [{}] for DN [{}]. ";
+
+    protected static final String ERROR_RETRIEVING_ATTRIBUTES_SEC_LOG = "Error retrieving attributes from external attribute store [%s] for DN [%s]. ";
+
+    private String attributeMapLocation;
+
+    private List<String> supportedClaims;
+
+    private Map<String, String> attributeMap;
+
+    protected SimpleSign simpleSign;
+
+    protected String externalAttributeStoreUrl;
+
+    protected String issuer;
+
+    protected String destination;
+
+    public AttributeQueryClaimsHandler() {
+        LOGGER.debug("Starting AttributeQueryClaimsHandler");
+    }
+
+    /**
+     * Gets the supported claim types.
+     *
+     * @return List of supported claims.
+     */
+    @Override
+    public List<URI> getSupportedClaimTypes() {
+        LOGGER.debug("Getting supported claim types.");
+        List<URI> supportedClaimTypes = new ArrayList<>();
+        try {
+            for (String claim : supportedClaims) {
+                supportedClaimTypes.add(new URI(claim));
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.warn("Not a valid URI for claim type {}.", e);
+        }
+        return supportedClaimTypes;
+    }
+
+    /**
+     * Retrieves claims from the external attribute store.
+     *
+     * @param claims The collection of claims.
+     * @return The collection of claims or null if there are no security claims.
```

Comment:
This will not return null since a new ProcessedClaimCollection will always be returned with or without the security attributes